libnetwork/libnetwork_internal_test.go

```diff
@@ -361,88 +358,6 @@ func TestAuxAddresses(t *testing.T) {
 	}
 }
 
-func TestUpdateSvcRecord(t *testing.T) {
-	skip.If(t, runtime.GOOS == "windows", "bridge driver and IPv6, only works on linux")
-
-	tests := []struct {
-		name       string
-		epName     string
-		addr4      string
-		addr6      string
-		expSvcRecs []etchosts.Record
-	}{
-		{
-			name:   "v4only",
-			epName: "ep4",
-			addr4:  "172.16.0.2/24",
-			expSvcRecs: []etchosts.Record{
-				{Hosts: "id-ep4", IP: netip.MustParseAddr("172.16.0.2")},
-			},
-		},
-		{
-			name:   "v6only",
-			epName: "ep6",
-			addr6:  "fde6:045d:b2aa::2/64",
-			expSvcRecs: []etchosts.Record{
-				{Hosts: "id-ep6", IP: netip.MustParseAddr("fde6:45d:b2aa::2")},
-			},
-		},
-		{
-			name:   "dual-stack",
-			epName: "ep46",
-			addr4:  "172.16.1.2/24",
-			addr6:  "fd60:8677:5a4c::2/64",
-			expSvcRecs: []etchosts.Record{
-				{Hosts: "id-ep46", IP: netip.MustParseAddr("172.16.1.2")},
-				{Hosts: "id-ep46", IP: netip.MustParseAddr("fd60:8677:5a4c::2")},
-			},
-		},
-	}
-
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			defer netnsutils.SetupTestOSContext(t)()
-			ctrlr, err := New(config.OptionDataDir(t.TempDir()))
-			assert.NilError(t, err)
-			defer ctrlr.Stop()
-
-			var ipam4, ipam6 []*IpamConf
-			var ip4, ip6 net.IP
-			if tc.addr4 != "" {
-				var net4 *net.IPNet
-				ip4, net4, err = net.ParseCIDR(tc.addr4)
-				assert.NilError(t, err)
-				ipam4 = []*IpamConf{{PreferredPool: net4.String()}}
-			}
-			if tc.addr6 != "" {
-				var net6 *net.IPNet
-				ip6, net6, err = net.ParseCIDR(tc.addr6)
-				assert.NilError(t, err)
-				ipam6 = []*IpamConf{{PreferredPool: net6.String()}}
-			}
-			n, err := ctrlr.NewNetwork("bridge", "net1", "", nil,
-				NetworkOptionEnableIPv4(tc.addr4 != ""),
-				NetworkOptionEnableIPv6(tc.addr6 != ""),
-				NetworkOptionIpam(defaultipam.DriverName, "", ipam4, ipam6, nil),
-			)
-			assert.NilError(t, err)
-			ep, err := n.CreateEndpoint(context.Background(), tc.epName,
-				CreateOptionDNSNames([]string{tc.epName, "id-" + tc.epName}),
-				CreateOptionIpam(ip4, ip6, nil, nil),
-			)
-			assert.NilError(t, err)
-
-			n.updateSvcRecord(context.Background(), ep, true)
```

Comment:
At a quick glance, I couldn't find another test covering `updateSvcRecord()`. I understand we may no longer be using `getSvcRecords()` now, but wondering if we should have a test covering it. Or should `updateSvcRecords` be replaced with something else in places where it's called?

Reply:
It's a fair cop! I started on a version of `getSvcRecords` for the test, convinced myself it was already covered by some other test, then couldn't find it again when I came to write the PR description. So, I've put it back - serves me right for not making notes!

(I added this test quite recently, in https://github.com/moby/moby/pull/48290/commits/344039b9ae771f9ac431baba583ff0ded3d5da80 - because that change modified `getSvcRecords` to deal with IPv6 "so that /etc/hosts entries are deleted when an IPv6-only endpoint is removed". That's no longer an issue of course.)